ftp/src/main/scala/akka/stream/alpakka/ftp/model.scala

```diff
@@ -255,18 +404,36 @@ sealed abstract class SftpIdentity {
  * @param privateKeyFilePassphrase password to use to decrypt private key
  * @param publicKey public key value to use when connecting
  */
-final case class RawKeySftpIdentity(
-    privateKey: Array[Byte],
-    privateKeyFilePassphrase: Option[Array[Byte]] = None,
-    publicKey: Option[Array[Byte]] = None
+final class RawKeySftpIdentity @InternalApi private[ftp] (
+    val privateKey: Array[Byte],
+    val privateKeyFilePassphrase: Option[Array[Byte]] = None,
+    val publicKey: Option[Array[Byte]] = None
 ) extends SftpIdentity {
   type KeyType = Array[Byte]
 
+  def withPrivateKey(value: KeyType): RawKeySftpIdentity = copy(privateKey = value)
+
   def withPrivateKeyFilePassphrase(privateKeyFilePassphrase: Array[Byte]): RawKeySftpIdentity =
     copy(privateKeyFilePassphrase = Some(privateKeyFilePassphrase))
 
   def withPublicKey(publicKey: KeyType): RawKeySftpIdentity =
     copy(publicKey = Some(publicKey))
+
+  private def copy(
+      privateKey: Array[Byte] = privateKey,
+      privateKeyFilePassphrase: Option[Array[Byte]] = privateKeyFilePassphrase,
+      publicKey: Option[Array[Byte]] = publicKey
+  ): RawKeySftpIdentity = new RawKeySftpIdentity(
+    privateKey,
+    privateKeyFilePassphrase,
+    publicKey
+  )
+
+  override def toString: String =
+    "RawKeySftpIdentity(" +
+    s"privateKey(length)=${privateKey.length}," +
+    s"privateKeyFilePassphrase=${privateKeyFilePassphrase.isDefined}," +
```

Comment:
Maybe the label should also include that we are printing the value of `isDefined` here, like: `privateKeyFilePassphrase.isDefined`

Reply:
When I looked into that, I felt the urge to improve the credentials creation experience altogether.